blog/2023-07-31-new-pre-post-command-plugin-hooks.mdx

```diff
@@ -0,0 +1,165 @@
+we provide more details on how and why you may use these to extend the default behavior of conda.
+
+<!-- truncate -->
+
+:::info
+
+To check out a fully functional example, check out the [conda-guard][conda-guard] project
+to see how these new plugin hooks are used in practice.
+:::
+
+
+---
+
+To explain how these plugin hooks can be used, we will cover two examples.
+
+## Conda guard and the "pre command" hook
+
+The project linked to above is called [`conda-guard`](https://github.com/travishathaway/conda-guard), and it extends conda by adding functionality to "guard"
+a conda environment from being changed. Sometimes users may want to do this to protect themselves from
+[unintentionally modifying their base environments][conda-issue].
+
+Using the "pre command" plugin hook, we can first specify which commands we want it to run for and then
+run a function that executes before the command gets executed. By doing this, we can figure out whether
+the environment is guarded or not and exit early if so. Below is how such a plugin hook would be defined:
+
+```python
+from conda.plugins import hookimpl, CondaPreCommand
+
+def conda_guard_pre_commands_action(command: str) -> None:
+    """Checks to see if the current environment is guarded"""
+    current_environment = get_current_environment()
+    
+    if is_guarded(current_environment):
+        raise CondaError(
+            f"Current environment is guarded. Run `conda guard {environment}` to "
+            "remove guard."
+        )
+
+@hookimpl
+def conda_pre_commands():
+    yield CondaPreCommand(
+        name=f"conda_guard_pre_command",
+        action=conda_guard_pre_commands_action,
+        run_for={"install", "remove", "update", "env_update", "env_remove"},
+    )
+```
+
+In the example above, we first register our plugin hook by defining a function called
+`conda_pre_commands` and then decorating it with the `hookimpl` function provided by conda.
+Inside the hook function itself we return a `CondaPreCommand` object, which defines the following:
+
+- **name**: how this plugin hook will be internally reference by conda
+- **action**: a callable object that will be invoked before the commands defined in `run_for` are
+  executed
+- **run_for**: a `set` of commands that this plugin hook should be used with
+
+Going back to our example, if we wanted to create a plugin that would prevent us from mistakenly
+modifying "guarded" environments, it would make sense to define `run_for` as all the conda commands
+that could potentially modify an environment (e.g. `install`, `remove`, etc.).
+
+To actually prevent us from modifying our guarded environments, the `action` function
+checks if the current environment is guarded and then throws a `CondaError` if it is to make
+sure that the program exits early with a meaningful error message for the user.
+
+:::info
+
+For the full implementation, please see the [conda-guard][conda-guard] project.
+
+:::
+
+## Simple analytics with "post command"
```

Comment:
```suggestion
## Simple command counter with the "post command" hook
```

Let's defuse this a little, analytics is a big word and we don't want to scare people away